### ISSUE TITLE
03-linear-regression-reg-demographic-health-data.ipynb

### ISSUE THREAD

**Issue by mezcolantriz** [4GeeksAcademy/CarlosRestrepo-machine-learning-python#3]
@carlos-villa-restrepo 

---

### **1. ERROR - Doble escalado (StandardScaler + MinMaxScaler)**

```python
# PRIMER ESCALADO: StandardScaler
norm_scaler = StandardScaler()
norm_scaler.fit(X_train[num_variables])
X_train[num_variables] = norm_scaler.transform(X_train[num_variables])
X_test[num_variables] = norm_scaler.transform(X_test[num_variables])

# SEGUNDO ESCALADO: MinMaxScaler SOBRE LOS YA ESCALADOS
minmax_scaler = MinMaxScaler()
minmax_scaler.fit(X_train[num_variables_escalar])  # ❌ Mismas variables
X_train[num_variables_escalar] = minmax_scaler.transform(X_train[num_variables_escalar])
X_test[num_variables_escalar] = minmax_scaler.transform(X_test[num_variables_escalar])
```

**PROBLEMA:**
- Aplicas **StandardScaler** a las variables (las normaliza: media=0, std=1)
- Luego aplicas **MinMaxScaler** a las **MISMAS variables ya normalizadas**
- Esto es **redundante y distorsiona los datos**

**Efecto del doble escalado:**
```python
# Datos originales: [50, 100, 150]

# Después de StandardScaler: [-1, 0, 1] (media=0, std=1)

# Después de MinMaxScaler sobre datos normalizados: [0, 0.5, 1]
# MinMaxScaler "normaliza" datos que ya estaban normalizados
# Resultado: Las variables no están en la escala esperada
```

**DEBERÍAS:**
- Elegir **UNO solo**: StandardScaler **O** MinMaxScaler
- Para regresión lineal regularizada, StandardScaler suele ser preferible
- Si quiere probar ambos, hacerlo en modelos separados, no secuencialmente

---

### **2. PROBLEMA - SelectKBest no se usa en los modelos**

```python
# Crea X_train_sel y X_test_sel con SelectKBest (k=5)
selection_model = SelectKBest(f_regression, k=5)
selection_model.fit(X_train, y_train)
X_train_sel = pd.DataFrame(...)
X_test_sel = pd.DataFrame(...)

# PERO LUEGO todos los modelos usan X_train y X_test completos
lr_model.fit(X_train, y_train)  # Usas X_train, no X_train_sel
lasso_model.fit(X_train, y_train)  # Usas X_train, no X_train_sel
ridge_model.fit(X_train, y_train)  # Usas X_train, no X_train_sel
```

**PROBLEMA:**
- Implementas SelectKBest correctamente para seleccionar 5 mejores features
- **Pero nunca usas** `X_train_sel` y `X_test_sel`
- Todos los modelos entrenan con el dataset completo
- El trabajo de selección de features es inútil

**DEBERÍAS:**
- Usar `X_train_sel` y `X_test_sel` en los modelos, **O**
- Eliminar el código de SelectKBest si no lo va a usar

---

### **3. NO CUMPLES REQUISITO - No analiza evolución R² vs alpha**

**Instrucciones del ejercicio:**
```markdown
Analiza cómo evoluciona el R² cuando el hiperparámetro del modelo Lasso cambia 
(puedes por ejemplo empezar a probar desde el valor 0.0 e ir aumentándolo hasta 
un valor de 20). Dibuja estos valores en un diagrama de líneas.
```

**Lo que haces:**
```python
lasso_model = Lasso(alpha=0.5, ...)  # Alpha fijo
lasso_cv_model = LassoCV(alphas=np.logspace(-6, 6, 10), cv=5, ...)  # CV automático
```

**OBSERVACIÓN:**
- LassoCV prueba 10 valores de alpha con validación cruzada ✅
- **PERO** no extraes ni visualiza la curva alpha vs R²
- No cumples completamente el requisito de dibujar el diagrama de líneas

---

### **4. Variables seleccionadas para scaling son solo 4**

```python
num_variables = ["MEDHHINC_2018", 
                 "Percent of adults with a bachelor's degree or higher 2014-18", 
                 "Percent of Population Aged 60+", 
                 "Unemployment_rate_2018"]
```

**OBSERVACIÓN:**
- El DataFrame tiene muchas más variables numéricas (después de limpieza tiene ~62 columnas)
- Solo escalas 4 variables manualmente seleccionadas
- Las otras variables numéricas quedan **sin escalar**
- Esto causa **inconsistencia de escala** entre variables

**DEBERÍAS:**
```python
# Seleccionar TODAS las numéricas automáticamente
num_variables = X_train.select_dtypes(include=['int64', 'float64']).columns.tolist()
```

---
